src/go/rpk/pkg/config/params.go

```diff
@@ -546,7 +546,14 @@ func (c *Config) addUnsetDefaults() {
 		brokers = []string{"127.0.0.1:9092"}
 	}
 
-	if len(r.AdminAPI.Addresses) == 0 {
+	addrs := r.AdminAPI.Addresses
+	defer func() { r.AdminAPI.Addresses = addrs }()
+	if len(addrs) == 0 && len(c.Redpanda.AdminAPI) > 0 {
+		for _, adminAPI := range c.Redpanda.AdminAPI {
+			addrs = append(addrs, net.JoinHostPort(adminAPI.Address, strconv.Itoa(adminAPI.Port)))
```

Comment:
This should probably have a preference about what listener to use.

The admin API can have mTLS on a per-listener basis: if someone has configured an insecure listener on 127.0.0.1, then that is probably the one that rpk wants to use.

Reply:
If we want to go this route, we can use https://pkg.go.dev/net#IP.IsPrivate and sort that one first.

Reply:
Check https://github.com/redpanda-data/redpanda/compare/f53770465bcc183c7108466d93f4799b73cb3cd0..1e169a554d00fc10045c90036a78e1eb635b56bd